error.go

```diff
@@ -45,27 +67,90 @@ func (e Errs) Error() string {
 	return fmt.Sprintf("%d error(s) occurred.\n\t%s", len(e), strings.Join(details, "\n\t"))
 }
 
-// Error returns a string containing the region, service as well as the original API error message.
-func (e *callErr) Error() string {
-	return fmt.Sprintf("region: %s, service: %s, Error message: %q",
-		e.region,
-		e.service,
-		e.err.Error())
-}
+// ErrorIn inspects err to find if there is an error in the region and returns it
+// or them (in case of multiple), otherwise it returns nil.
+// err must be of the type Error or Errors in order to be able to find if there
+// are errors in the region; if err is from other type, the function always
+// returns nil.
+// The returned error is a value of the type Error when only one error is found
+// in the region, or a value of the type Errors when multiple errors are found.
+func ErrorIn(region string, err error) error {
+	switch e := err.(type) {
+	case Error:
+		if e.Region() == region {
+			return err
+		}
 
-// Returns the region of the error
-func (e *callErr) Region() string {
-	return e.region
-}
+		return nil
+	case *Error:
+		if e == nil {
+			return nil
+		}
 
-// Returns the service name of the error
-func (e *callErr) Service() string {
-	return e.service
+		if e.Region() == region {
+			return err
+		}
+
+		return nil
+	case Errors:
+		var errs Errors
+
+		for _, ei := range e {
+			if ei.Region() == region {
+				errs = append(errs, ei)
+			}
```

Comment:
I would use recursion for this, just call `ErrorIn(region, e)` and append if not nil. Would make the code more reusable, this way you only have the logic on one `case`, the `Error`

Reply:
Same than mentioned in https://github.com/cycloidio/raws/pull/25#discussion_r166856366